frontend/catalyst/ag_primitives.py

```diff
@@ -278,12 +295,28 @@ def for_stmt(
         try:
             set_state(init_state)
             results = _call_catalyst_for(
-                start, stop, step, body_fn, get_state, set_state, opts, enum_start, iteration_array
+                start,
+                stop,
+                step,
+                body_fn,
+                get_state,
+                set_state,
+                symbol_names,
+                enum_start,
+                iteration_array,
             )
+
         except Exception as e:  # pylint: disable=broad-exception-caught
             if catalyst.autograph_strict_conversion:
                 raise e
 
+            if EvaluationContext.is_tracing():
```

Comment:
Could you maybe add a small comment on why we try to retrieve the Jax PR frame?

Reply:
done: https://github.com/PennyLaneAI/catalyst/pull/352/commits/e9ccbcb0206a62ef393033ba759a334443d5e49e